migration/identity-migration/src/main/java/io/camunda/migration/identity/config/ManagementIdentityConnectorConfig.java

```diff
@@ -110,12 +110,14 @@ public class EndpointUnavailableInterceptor implements ClientHttpRequestIntercep
     public ClientHttpResponse intercept(
         final HttpRequest request, final byte[] body, final ClientHttpRequestExecution execution)
         throws IOException {
-      final var result = execution.execute(request, body);
-      if (result.getStatusCode() == HttpStatus.NOT_FOUND) {
+      final var response = execution.execute(request, body);
+      if (response.getStatusCode() == HttpStatus.NOT_FOUND) {
+        // the response needs to be closed as it won't be processed anymore due the exception thrown
+        response.close();
```

Comment:
this is the actual fix, the dedicated IT class `KeycloakIdentityMigrationNoRbaNoMultiTenancyIT` ensures we have the exact setup needed to trigger the leakage, being keycloak with RBA and MT disabled + at least 5 groups (without authorizations)